Mirador

flummingbird opened the pull request:
**GitHub Issue**: (https://github.com/Islandora-CLAW/CLAW/issues/645)

# What does this Pull Request do?

downloads the drupal 8 mirador module, which is note based off of core mirador but rather an obscure drupal friendly branch. Also pulls the mirador javascript dependency.

Feedback todos from last pull request have been completed.
Downloads and installs mirador drupal8 module, and JavaScript dependencies.



# What's new?

permission issues fixed thanks to @ruebot @whikloj and @dannylamb 

* Added scripts/mirador.sh

* Could this change impact execution of existing code?
no

# How should this be tested?

* To Test that the Pull Request does what is intended:

clone the repository

cd into repository folder

vagrant up

check localhost:8000/admin/config/media/mirador


# Additional Notes:
I branched my own fork, I'm not sure if I can create a branch on the offical claw_vagrant to merge my PR into. Suggestions?

# Interested parties
@Islandora-CLAW/committers @whikloj @ruebot @dannylamb

flummingbird committed:
initial commit, pretty sure that I did many things "wrong.." but it skips a few steps of mirador install for you.

flummingbird committed:
Merge branch 'master' of https://github.com/flummingbird/claw_vagrant

flummingbird committed:
Update Vagrantfile

added needed line to Vagrant file for /scripts/mirador.sh currently commented out bc I haven't tested a clean build of all this together.

flummingbird committed:
fullfill PR feedback tasks

flummingbird committed:
remove comment per feedback

whikloj commented:
Ahhh I didn't realize there was a new PR. This supercedes #41.

whikloj commented:
@flummingbird so the only thing I would suggest is to perhaps take a look at [this section](https://github.com/Islandora-CLAW/claw_vagrant/blob/master/scripts/fcrepo.sh#L10-L13) of another script.

We download into a specific location and _only_ if the file doesn't exist.

That way if you do a `vagrant destroy; vagrant up` and the versions haven't changed you don't need to re-download those files. We just copy them into the `/tmp` directory (for example) and move ahead.

Reduces time on rebuilding and allows you to build a machine while offline.

flummingbird committed:
Update mirador.sh

conditional download and enabling, prevents repeated downloads, and uses proper $DOWNLOAD_DIR variable. I have tested it locally and it works like a charm.

flummingbird reviewed (commented):
I've made some changes, I believe it is ready.

flummingbird committed:
Update mirador.sh

uncouple download to wait for each check. Enable and unzip go outside the conditional downloads.

flummingbird committed:
Update mirador.sh

puts unzip back in conditional check.

flummingbird committed:
Update Vagrantfile

whikloj commented on scripts/mirador.sh:
```diff
@@ -0,0 +1,24 @@
+if [ ! -f "$DOWNLOAD_DIR/master.zip" ]; then
+    echo "Downloading mirador-js dependencies"
+    cd "$DOWNLOAD_DIR" && { curl -LOk https://github.com/NVLI/mirador-js/archive/master.zip; cd -;}
+    cd "$DRUPAL_HOME"/web/libraries && { unzip "$DOWNLOAD_DIR"/master.zip -d . ;}
```
@flummingbird this line needs to move outside the `if...fi` (line 16) as it should occur whenever a box is provisioned and is separate from the downloading. 
The way I think of it is that lines 10-12, is just checking if you have downloading the application before. 
Then line 13 is deploying that application into your vagrant machine.

So you can conceivably run lines 10-12 once and use line 13 waaaaaay more than that.

whikloj commented on scripts/mirador.sh:
```diff
@@ -0,0 +1,24 @@
+if  [ ! -f "$DOWNLOAD_DIR/mirador-8.x-1.x-dev.zip" ]; then
+    echo "Downolading mirador drupal module"
+    cd "$DOWNLOAD_DIR" && { curl -LOk https://ftp.drupal.org/files/projects/mirador-8.x-1.x-dev.zip ;}
+    cd "$DRUPAL_HOME"/web/modules/contrib/ && { unzip "$DOWNLOAD_DIR"/mirador-8.x-1.x-dev.zip -d . ;}
```
Same thing as described [above](https://github.com/Islandora-CLAW/claw_vagrant/pull/42/files#r120160162)

whikloj commented on scripts/mirador.sh:
```diff
@@ -0,0 +1,24 @@
+fi
+
+if  [ ! -f "$DOWNLOAD_DIR/mirador-8.x-1.x-dev.zip" ]; then
+    echo "Downolading mirador drupal module"
```
typo `Downolading`

flummingbird replied:
phooey

flummingbird committed:
Update mirador.sh

typey type-OH

flummingbird committed:
Update mirador.sh

unzips outside of DL function.

flummingbird commented:
:man_dancing:

whikloj commented:
@flummingbird just wanted to let you know, I haven't forgotten about this. I think due to the Mirador module only working against the `image` type and image not allowing you to upload anything that isn't a gif, jpeg, or png... we need to hold this until 8.4.x is released and we move to it.
